### PR TITLE
Hopefully this makes me go into team. :)

### PR DESCRIPTION
**Pull request by sayantan300**
All the images in your repo has been optimized ;)
Also add ImgBot app to this repo so you can optimize new images automatically.
Ps: I'm starting the work again on dmod.
To-do:
- search
- exposing api
- fix user schema (I found some bugs with mine)

### CODE REVIEW

**Comment by sayantan300** [comment]
@MattA-Official I gave some fixes, also just asking you some stuff.
Why are you sending 200 status? Expressjs or other http routers send it automatically if there's no status given.

Also please change my javascript code to typescript (const...require) that part.

I'm hopefully learning typescript I think.

**Comment by sayantan300** [comment]
Ps I'm going to work on search as soon as this is merged.

File: pages/api/v1/users/[id]/isMod.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+var Users = require("../../../../models/users.ts"); /*i think this correctly requires the users model*/
+
+export default async (req: NextApiRequest, res: NextApiResponse) => {
+ const user = await Users.findOne({id: req.params.id});
+	 await res.json(user); //dont send as "user: {}" but simply {}
```
Comment (MattA-Official) [inline pages/api/v1/users/[id]/isMod.ts:6]:
```suggestion
	 await res.status(200).json(user); //dont send as "user: {}" but simply {}
```

File: pages/api/index.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import type { NextApiRequest, NextApiResponse } from 'next'
 
 export default (req: NextApiRequest, res: NextApiResponse) => {
-	res.status(200).json({ hello: 'world' })
+	res.json({ hello: 'world' })
```
Comment (MattA-Official) [inline pages/api/index.ts:4]:
```suggestion
	res.status(200).json({ hello: 'world' })
```

File: pages/api/v1/users/[id]/isMod.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+var Users = require("../../../../models/users.ts"); /*i think this correctly requires the users model*/
```
Comment (MattA-Official) [inline pages/api/v1/users/[id]/isMod.ts:2]:
```suggestion
import Users from "../../../../models/users.ts"; /*i think this correctly requires the users model*/
```

File: pages/api/v1/users/[id].ts [hunk 4]
```diff
@@ -10,15 +11,17 @@ export default (req: NextApiRequest, res: NextApiResponse) => {
 
 	switch (method) {
 		case 'GET':
-			res.status(200).json({ id })
+			const user = await Users.findOne({id: req.params.id});
+			await res.json(user);
```
Comment (MattA-Official) [inline pages/api/v1/users/[id].ts:15]:
```suggestion
			await res.status(200).json(user);
```

File: pages/api/v1/users/[id].ts [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import type { NextApiRequest, NextApiResponse } from 'next'
+var Users = require("../../../models/users.ts"); /*i think this correctly requires the users model*/
```
Comment (MattA-Official) [inline pages/api/v1/users/[id].ts:2]:
```suggestion
import Users from "../../../models/users.ts"; /*i think this correctly requires the users model*/
```

**Review by MattA-Official — CHANGES_REQUESTED** [review]

File: pages/api/v1/users/[id]/isMod.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+import Users from "../../../../models/users.ts"; /*i think this correctly requires the users model*/
+
+export default async (req: NextApiRequest, res: NextApiResponse) => {
+ const user = await Users.findOne({id: req.params.id});
```
Comment (MattA-Official) [inline pages/api/v1/users/[id]/isMod.ts:5]:
```suggestion
 const user = await Users.findOne({id: req.query.id});
```

File: pages/api/v1/users/[id].ts [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import type { NextApiRequest, NextApiResponse } from 'next'
+import Users from "../../../models/users.ts"; /*i think this correctly requires the users model*/
```
Comment (MattA-Official) [inline pages/api/v1/users/[id].ts:2]:
```suggestion
import Users from "../../../models/users"; /*i think this correctly requires the users model*/
```

File: pages/api/v1/users/[id]/isMod.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+import Users from "../../../../models/users.ts"; /*i think this correctly requires the users model*/
```
Comment (MattA-Official) [inline pages/api/v1/users/[id]/isMod.ts:2]:
```suggestion
import Users from "../../../../models/users"; /*i think this correctly requires the users model*/
```

**Review by MattA-Official — CHANGES_REQUESTED** [review]

File: pages/api/v1/users/[id].ts [hunk 9]
```diff
@@ -10,15 +11,17 @@ export default (req: NextApiRequest, res: NextApiResponse) => {
 
 	switch (method) {
 		case 'GET':
-			res.status(200).json({ id })
+			const user = await Users.findOne({id: req.params.id});
```
Comment (MattA-Official) [inline pages/api/v1/users/[id].ts:14]:
```suggestion
			const user = await Users.findOne({id: req.query.id});
```

File: pages/api/v1/users/[id]/isMod.ts (hunk 6 above)
Comment (sayantan300) [reply pages/api/v1/users/[id]/isMod.ts:5]: @MattA-Official uh isn't `[id]` folder resolve to req.params.id?
I'm sure it does. Probably you're forgetting how to use express and other servers...

**Comment by sayantan300** [comment]
@MattA-Official you did fully wrong. Lemme fix it myself. Also, I'm removing ".ts" extension for import.

**Comment by sayantan300** [comment]
Actually you just only remove .ts extension.
Cause I don't know how to make this back into draft pull request.

File: pages/api/v1/users/[id]/isMod.ts (hunk 6 above)
Comment (MattA-Official) [reply pages/api/v1/users/[id]/isMod.ts:5]: > @MattA-Official uh isn't `[id]` folder resolve to req.params.id?
> I'm sure it does. Probably you're forgetting how to use express and other servers...

With NextJS it is `query`, not `param`. This was pointed out by an automatic type check

**Comment by sayantan300** [comment]
@MattA-Official ... I asked you not to use `req.query`

File: pages/api/v1/users/[id].ts [hunk 10]
```diff
@@ -1,6 +1,7 @@
 import type { NextApiRequest, NextApiResponse } from 'next'
+import Users from "../../../models/users"; /*i think this correctly requires the users model*/
```
Comment (MattA-Official) [inline pages/api/v1/users/[id].ts:2]:
```suggestion
import Users from "../../../../../models/users"; /*i think this correctly requires the users model*/
```

File: pages/api/v1/users/[id]/isMod.ts [hunk 11]
```diff
@@ -0,0 +1,8 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+import Users from "../../../../models/users"; /*i think this correctly requires the users model*/
```
Comment (MattA-Official) [inline pages/api/v1/users/[id]/isMod.ts:2]:
```suggestion
import Users from "../../../../../../models/users"; /*i think this correctly requires the users model*/
```

**Review by MattA-Official — CHANGES_REQUESTED** [review]

File: pages/api/v1/users/[id].ts [hunk 12]
```diff
@@ -1,6 +1,7 @@
 import type { NextApiRequest, NextApiResponse } from 'next'
+import Users from "../../../../../models/users"; /*i think this correctly requires the users model*/
```
Comment (MattA-Official) [inline pages/api/v1/users/[id].ts:2]:
```suggestion
import Users from "@models/users"; /*i think this correctly requires the users model*/
```

File: pages/api/v1/users/[id]/isMod.ts [hunk 13]
```diff
@@ -0,0 +1,8 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+import Users from "../../../../../../models/users"; /*i think this correctly requires the users model*/
```
Comment (MattA-Official) [inline pages/api/v1/users/[id]/isMod.ts:2]:
```suggestion
import Users from "@models/users"; /*i think this correctly requires the users model*/
```

**Review by MattA-Official — CHANGES_REQUESTED** [review]

**Pull request merged**